### ISSUE TITLE
Issue with dll dependencies

### ISSUE THREAD

**Issue by YannRobert** [schwabdidier/GazePlay#161]
On Windows 10, (fresh install)
using Tobii 4C with latest drivers

GazePlay fails with following error in logs

```
2018-01-20 19:35:42,463 ERROR - Exception in thread "JavaFX Application Thread" java.lang.UnsatisfiedLinkError: C:\Users\yann\AppData\Local\Temp\lib\tobii\x64\tobii_jni_stream_engine.dll: Can't find dependent libraries
2018-01-20 19:35:42,464 ERROR - 	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
2018-01-20 19:35:42,464 ERROR - 	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
2018-01-20 19:35:42,464 ERROR - 	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
2018-01-20 19:35:42,464 ERROR - 	at java.lang.Runtime.load0(Runtime.java:809)
2018-01-20 19:35:42,464 ERROR - 	at java.lang.System.load(System.java:1086)
2018-01-20 19:35:42,464 ERROR - 	at tobii.Tobii.loadLibraryFromResources(Tobii.java:46)
2018-01-20 19:35:42,464 ERROR - 	at tobii.Tobii.<init>(Tobii.java:24)
2018-01-20 19:35:42,464 ERROR - 	at tobii.Tobii.gazePosition(Tobii.java:14)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.commons.gaze.GazeTobii.execProg(GazeTobii.java:31)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.commons.gaze.GazeUtils.createGazeListener(GazeUtils.java:40)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.commons.gaze.GazeUtils.<clinit>(GazeUtils.java:26)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.commons.utils.stats.Stats.<init>(Stats.java:72)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.commons.utils.stats.ShootGamesStats.<init>(ShootGamesStats.java:14)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.games.creampie.CreampieStats.<init>(CreampieStats.java:9)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.DefaultGamesLocator$1.createNewStats(DefaultGamesLocator.java:35)
2018-01-20 19:35:42,464 ERROR - 	at net.gazeplay.HomeMenuScreen.chooseGame(HomeMenuScreen.java:198)
```

**Comment by schwabdidier** [comment]
Fech as we say in French. It works perfectly well on our three windows computers... Why do we have this problem ??

**Comment by schwabdidier** [comment]
One of our tester have exactly the same error as you @YannRobert ...

**Comment by YannRobert** [comment]
Would it be possible to document the dependencies ?

**Comment by schwabdidier** [comment]
On a computer where the 4C works, we have

`java.library.path: C:\ProgramData\Oracle\Java\javapath;C:\Windows\Sun\Java\bin;C:\Windows\system32;C:\Windows;C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Git\cmd;`

On a computer where the 4C DOESN'T work, we have
` java.library.path: C:\Program Files\Java\jdk1.8.0_161\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Users\<UserName>\AppData\Local\Microsoft\WindowsApps;.`

**Comment by coylz** [comment]
@YannRobert @schwabdidier 
I think that everything is fixed in the TobiiStreamEngineForJava project now.
I tested on a VM freshly installed without anything else than Java and it worked.

We need one last thing to do on the GazePlay side : add the directory `%LocalAppData%\TobiiStreamEngineForJava\lib\tobii\x64` to the `PATH` environment variable in the `.bat` :)
For instance we could add the line "PATH=%LocalAppData%\TobiiStreamEngineForJava\lib\tobii\x64;%PATH%" just before the "java ..." line.

**Comment by YannRobert** [comment]
@coylz very good

I had a different idea that I exposed a few days ago in an email.
that was to create the dll directory at build time in the gazeplay-dist.zip, so that it already contains the dll, so that the java program does not need to copy the dll files from the jar to the directory at runtime.

This will be better in terms of performances, in terms of avoiding corrupted files if the java program is killed during the file copy, also the java program will be much simpler. Also, it will be easier for anyone to add or change a dll file manually (either because we as a developer need to experiment something, or because any user wants to use a more recent dll)

**Comment by schwabdidier** [comment]
Maybe linked to @YannRobert's comment

> 2018-01-31 17:25:51,562 INFO  - Eye-tracker = tobii_eyeX_4C
> 2018-01-31 17:25:51,566 INFO  - Tobii: Loading needed libraries using directory C:\Users\schwab\AppData\Local/TobiiStreamEngineForJava
> 2018-01-31 17:25:51,566 INFO  - Tobii: Copying /lib/tobii/x64/vcruntime140.dll into C:\Users\schwab\AppData\Local/TobiiStreamEngineForJava
> 2018-01-31 17:25:51,579 INFO  - Tobii: Copying /lib/tobii/x64/msvcp140.dll into C:\Users\schwab\AppData\Local/TobiiStreamEngineForJava
> 2018-01-31 17:25:51,589 INFO  - Tobii: Copying /lib/tobii/x64/concrt140.dll into C:\Users\schwab\AppData\Local/TobiiStreamEngineForJava
> 2018-01-31 17:25:51,596 INFO  - Tobii: Copying /lib/tobii/x64/vccorlib140.dll into C:\Users\schwab\AppData\Local/TobiiStreamEngineForJava
> 2018-01-31 17:25:51,602 INFO  - Tobii: Copying /lib/tobii/x64/tobii_stream_engine.dll into C:\Users\schwab\AppData\Local/TobiiStreamEngineForJava
> 2018-01-31 17:25:51,618 INFO  - Tobii: Copying /lib/tobii/x64/tobii_jni_stream_engine.dll into C:\Users\schwab\AppData\Local/TobiiStreamEngineForJava
> 2018-01-31 17:25:51,622 INFO  - Tobii: Loading library C:\Users\schwab\AppData\Local\TobiiStreamEngineForJava\lib\tobii\x64\tobii_jni_stream_engine.dll
> 2018-01-31 17:25:51,626 ERROR - java.lang.UnsatisfiedLinkError: C:\Users\schwab\AppData\Local\TobiiStreamEngineForJava\lib\tobii\x64\tobii_jni_stream_engine.dll: Can't find dependent libraries
> 2018-01-31 17:25:51,626 ERROR - 	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
> 2018-01-31 17:25:51,626 ERROR - 	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
> 2018-01-31 17:25:51,627 ERROR - 	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
> 2018-01-31 17:25:51,627 ERROR - 	at java.lang.Runtime.load0(Runtime.java:809)
> 2018-01-31 17:25:51,627 ERROR - 	at java.lang.System.load(System.java:1086)
> 2018-01-31 17:25:51,627 ERROR - 	at tobii.Tobii.loadLibrary(Tobii.java:78)
> 2018-01-31 17:25:51,627 ERROR - 	at tobii.Tobii.loadNeededLibraries(Tobii.java:61)
> 2018-01-31 17:25:51,627 ERROR - 	at tobii.Tobii.loadIfNotLoaded(Tobii.java:40)
> 2018-01-31 17:25:51,627 ERROR - 	at tobii.Tobii.gazePosition(Tobii.java:22)
> 2018-01-31 17:25:51,627 ERROR - 	at net.gazeplay.commons.gaze.devicemanager.TobiiGazeDeviceManager.init(TobiiGazeDeviceManager.java:22)
> 2018-01-31 17:25:51,627 ERROR - 	at net.gazeplay.commons.gaze.devicemanager.GazeDeviceManagerFactory.createNewGazeListener(GazeDeviceManagerFactory.java:53)
> 2018-01-31 17:25:51,627 ERROR - 	at net.gazeplay.GameContext.newInstance(GameContext.java:83)
> 2018-01-31 17:25:51,627 ERROR - 	at net.gazeplay.HomeMenuScreen.chooseGame(HomeMenuScreen.java:195)
> 2018-01-31 17:25:51,627 ERROR - 	at net.gazeplay.HomeMenuScreen.chooseGame(HomeMenuScreen.java:187)
> 2018-01-31 17:25:51,628 ERROR - 	at net.gazeplay.HomeMenuScreen.access$000(HomeMenuScreen.java:42)
> 2018-01-31 17:25:51,628 ERROR - 	at net.gazeplay.HomeMenuScreen$1.changed(HomeMenuScreen.java:157)
> 2018-01-31 17:25:51,628 ERROR - 	at net.gazeplay.HomeMenuScreen$1.changed(HomeMenuScreen.java:154)
> 2018-01-31 17:25:51,628 ERROR - 	at com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(ExpressionHelper.java:361)
> 2018-01-31 17:25:51,628 ERROR - 	at com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:81)
> 2018-01-31 17:25:51,628 ERROR - 	at javafx.beans.property.ReadOnlyIntegerPropertyBase.fireValueChangedEvent(ReadOnlyIntegerPropertyBase.java:72)
> 2018-01-31 17:25:51,628 ERROR - 	at javafx.beans.property.ReadOnlyIntegerWrapper.fireValueChangedEvent(ReadOnlyIntegerWrapper.java:102)
> 2018-01-31 17:25:51,628 ERROR - 	at javafx.beans.property.IntegerPropertyBase.markInvalid(IntegerPropertyBase.java:113)
> 2018-01-31 17:25:51,628 ERROR - 	at javafx.beans.property.IntegerPropertyBase.set(IntegerPropertyBase.java:147)
> 2018-01-31 17:25:51,628 ERROR - 	at javafx.scene.control.SelectionModel.setSelectedIndex(SelectionModel.java:68)
> 2018-01-31 17:25:51,628 ERROR - 	at javafx.scene.control.SingleSelectionModel.updateSelectedIndex(SingleSelectionModel.java:215)
> 2018-01-31 17:25:51,629 ERROR - 	at javafx.scene.control.SingleSelectionModel.select(SingleSelectionModel.java:149)
> 2018-01-31 17:25:51,629 ERROR - 	at javafx.scene.control.ChoiceBox$ChoiceBoxSelectionModel.select(ChoiceBox.java:573)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.scene.control.skin.ChoiceBoxSkin.lambda$addPopupItem$284(ChoiceBoxSkin.java:261)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
> 2018-01-31 17:25:51,629 ERROR - 	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
> 2018-01-31 17:25:51,630 ERROR - 	at javafx.event.Event.fireEvent(Event.java:198)
> 2018-01-31 17:25:51,630 ERROR - 	at javafx.scene.control.MenuItem.fire(MenuItem.java:462)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.doSelect(ContextMenuContent.java:1405)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.lambda$createChildren$343(ContextMenuContent.java:1358)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
> 2018-01-31 17:25:51,630 ERROR - 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
> 2018-01-31 17:25:51,631 ERROR - 	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
> 2018-01-31 17:25:51,631 ERROR - 	at javafx.event.Event.fireEvent(Event.java:198)
> 2018-01-31 17:25:51,631 ERROR - 	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
> 2018-01-31 17:25:51,631 ERROR - 	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
> 2018-01-31 17:25:51,632 ERROR - 	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
> 2018-01-31 17:25:51,632 ERROR - 	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:381)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
> 2018-01-31 17:25:51,632 ERROR - 	at java.security.AccessController.doPrivileged(Native Method)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$354(GlassViewEventHandler.java:417)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:416)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.glass.ui.View.notifyMouse(View.java:937)
> 2018-01-31 17:25:51,632 ERROR - 	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
> 2018-01-31 17:25:51,633 ERROR - 	at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
> 2018-01-31 17:25:51,633 ERROR - 	at java.lang.Thread.run(Thread.java:745)
> 2018-01-31 17:25:51,755 INFO  - Properties loaded : {FILEDIR=C:\Users\schwab\GazePlay\files\, WHEREISITDIR=, EYETRACKER=tobii_eyeX_4C, LANGUAGE=fra, FIXATIONLENGTH=500, CSSFILE=data/stylesheets/main-green.css, QUESTIONLENGTH=5000}
> 2018-01-31 17:25:51,757 INFO  - [data/stylesheets/main-green.css]
> 2018-01-31 17:25:51,757 INFO  - Finished setup stage with the game scene
> 2018-01-31 17:25:51,758 INFO  - computeHeatMapPixelSize() : result = 3.2
> 2018-01-31 17:25:51,760 INFO  - heatMapWidth = 320, heatMapHeight = 400
> 2018-01-31 17:25:51,821 INFO  - Properties loaded : {FILEDIR=C:\Users\schwab\GazePlay\files\, WHEREISITDIR=, EYETRACKER=tobii_eyeX_4C, LANGUAGE=fra, FIXATIONLENGTH=500, CSSFILE=data/stylesheets/main-green.css, QUESTIONLENGTH=5000}
> 2018-01-31 17:25:51,821 INFO  - filesFolder : C:\Users\schwab\GazePlay\files\
> 2018-01-31 17:25:51,822 INFO  - Try to find images in folder : C:\Users\schwab\GazePlay\files\\images\portraits
> 2018-01-31 17:25:51,822 INFO  - Folder doesn't exist : C:\Users\schwab\GazePlay\files\\images\portraits
> 2018-01-31 17:25:52,478 INFO  - result = Dimension2D [width = 1280.0, height = 984.0]
> 2018-01-31 17:25:52,479 INFO  - nodesEventFilterListSize = 1
> 2018-01-31 17:26:01,517 INFO  - Try to play data/creampie/sounds/missile.mp3
> 2018-01-31 17:26:01,518 INFO  - path file:/C:/Users/schwab/IdeaProjects/GazePlay/gazeplay-data/target/classes/data/creampie/sounds/missile.mp3
> 2018-01-31 17:26:01,795 INFO  - result = Dimension2D [width = 1280.0, height = 910.0]
> 2018-01-31 17:26:07,828 INFO  - stats = Stats{nbShoots = 1, length = 9035, average length = 9035, zero time = 16070}[9035]
> 2018-01-31 17:26:07,832 ERROR - Exception on Gaze position update
> java.lang.InterruptedException: sleep interrupted
> 	at java.lang.Thread.sleep(Native Method)
> 	at net.gazeplay.commons.gaze.devicemanager.TobiiGazeDeviceManager$1$1.call(TobiiGazeDeviceManager.java:50)
> 	at net.gazeplay.commons.gaze.devicemanager.TobiiGazeDeviceManager$1$1.call(TobiiGazeDeviceManager.java:32)
> 	at javafx.concurrent.Task$TaskCallable.call(Task.java:1423)
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> 	at javafx.concurrent.Service.lambda$null$492(Service.java:725)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at javafx.concurrent.Service.lambda$executeTask$493(Service.java:724)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
> 	at java.lang.Thread.run(Thread.java:745)

**Comment by coylz** [comment]
Did you add the path to the PATH ? :p

**Comment by schwabdidier** [comment]
I was unable to run it. Too tired to continue !

**Comment by YannRobert** [comment]
In order to be able to run gazeplay in your IDE, 
without having to mess around your global PATH env variable for the Windows user.

Just configure que run configuration of your IDE, in order to add the dll directory to the Path env variable.

The value to set would be
```
C:\Users\yann\AppData\Local\TobiiStreamEngineForJava\lib\tobii\x64
```
if your windows username is ```yann``` (adapt to your own username)
(did not managed to use a environment variable reference here, it seems you have to use the fully hardcoded value)

Here is a screenshot showing how to do it with Intellij

![intellij_run_configuration_screenshot](https://user-images.githubusercontent.com/1653590/35654449-7125e424-06ed-11e8-847e-3f3a5c5169a9.png)

**Comment by YannRobert** [comment]
fixed in 1.2
